gesturecontrol.py

```diff
@@ -48,6 +47,11 @@ def grabVector(self):
 
 		return (vectorX, vectorY, vectorZ)
 
+	def updateVectors(self, vectorX, vectorY, vectorZ):
+		self.vectorX = vectorX
+		self.vectorY = vectorY
+		self.vectorZ = vectorZ
+
```

Comment:
Place the code `if self.vectorLock.acquire(False):`  here.

Don't forget to release the lock with `self.vectorLock.release()`